### PR TITLE
Generate docs using pdocs python package

### PR DESCRIPTION
**Pull request by adityabharti**
PR checklist:

- [ #21 ] Github Issue associated with branch.
- [ NA ] Code passes linting through autopep8.
- [ Yes] PR branch is up-to-date with latest master branch.

Sample gh page link: https://adityabharti.github.io/fawkes/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,8 +16,52 @@ jobs:
...
+          paths: project
+
+  docs-deploy:
+      docker:
+      - image: node:8.10.0
+      steps:
+      - checkout
+      - add_ssh_keys:
+            fingerprints:
+                - "7f:f7:8d:79:e6:17:7a:4c:e3:ec:b7:05:55:57:fe:26"
```
Comment (MadaraUchiha-314) [inline .circleci/config.yml:45]: You should not add your fingerprints like this to the repo. Ideally all secrets should be stored as environment variables in the CICD platform (CircleCI here)

File: .circleci/config.yml [hunk 2]
```diff
@@ -16,8 +16,52 @@ jobs:
...
+      - attach_workspace:
+            at: ~/project/docs/project
+      - run:
+            name: Install and configure dependencies
+            command: |
+                npm install -g --silent gh-pages@2.0.1
+                git config user.email "aditya.9395@gmail.com"
```
Comment (MadaraUchiha-314) [inline .circleci/config.yml:52]: Kindly do not add your own personal emails here.

File: .circleci/config.yml (hunk 1 above)
Comment (adityabharti) [reply .circleci/config.yml:45]: Quoting [this article](https://circleci.com/blog/deploying-documentation-to-github-pages-with-continuous-integration/) from the CircleCI website:

"fingerprint for our key, which is a unique identifier that’s safe to expose publicly (unlike the private key itself..."

Let me know if we still need to add an environment variable.

File: .circleci/config.yml (hunk 2 above)
Comment (adityabharti) [reply .circleci/config.yml:52]: A valid git credential is needed to push the generated docs from the CircleCI containers to the gh-pages branch. Let me know how/what to replace it with.

https://circleci.com/docs/2.0/gh-bb-integration/#deployment-keys-and-user-keys

Comment (MadaraUchiha-314) [reply .circleci/config.yml]: Yes. Working on it.

File: .circleci/config.yml (hunk 1 above)
Comment (MadaraUchiha-314) [reply .circleci/config.yml]: By whose finger print is this ? And why should it be added to this repo....that's the question.

Comment (adityabharti) [reply .circleci/config.yml:45]: This is actually "CircelCI's fingerprint" but was added in my organisation. This fingerprint is generated when the private key is handed over to CircleCI. This needs to be done for Intuit organisation. [Steps here.](https://circleci.com/docs/2.0/gh-bb-integration/#deployment-keys-and-user-keys)

Also, for some reason build failed when I tried to replace this with env variable on CircleCI. When specified back directly, the build passed. We can update this later when Intuit org has created a fingerprint.

**Comment by MadaraUchiha-314** [comment]
@adityabharti Can you use https://intuit.github.io/auto/docs/configuration/non-npm along with the gh-pages plugin provided by auto ?

I can provide the email address and the username as environment variables so that it can be picked up.

**Comment by MadaraUchiha-314** [comment]
@adityabharti Are you going to make the changes required or should I take this forward ?

**Comment by adityabharti** [comment]
@MadaraUchiha-314 You may take this forward, been running a little busy. Thanks!

**Pull request closed without merging**